client/beefy/src/worker.rs

```diff
@@ -217,23 +218,28 @@ where
 			// TODO: (adoerr) Enacting a new authority set will also implicitly 'conclude'
 			// the currently active BEEFY voting round by starting a new one. This is
 			// temporary and needs to be replaced by proper round life cycle handling.
-			if active.id != self.rounds.validator_set_id() ||
-				(active.id == GENESIS_AUTHORITY_SET_ID && self.best_beefy_block.is_none())
+			if self.rounds.is_none() ||
+				active.id() != self.rounds.as_ref().unwrap().validator_set_id() ||
+				(active.id() == GENESIS_AUTHORITY_SET_ID && self.best_beefy_block.is_none())
 			{
 				debug!(target: "beefy", "🥩 New active validator set id: {:?}", active);
-				metric_set!(self, beefy_validator_set_id, active.id);
+				metric_set!(self, beefy_validator_set_id, active.id());
 
 				// BEEFY should produce a signed commitment for each session
-				if active.id != self.last_signed_id + 1 && active.id != GENESIS_AUTHORITY_SET_ID {
+				if active.id() != self.last_signed_id + 1 && active.id() != GENESIS_AUTHORITY_SET_ID
+				{
 					metric_inc!(self, beefy_skipped_sessions);
 				}
 
-				// verify the new validator set
-				let _ = self.verify_validator_set(notification.header.number(), active.clone());
+				if log_enabled!(log::Level::Debug) {
```

Comment:
```suggestion
				if log_enabled!(target: "beefy", log::Level::Debug) {
```

Since that's the override target that the `debug` statement oyu are referring in the comment uses. Otherwise it's unlikely to ever run.